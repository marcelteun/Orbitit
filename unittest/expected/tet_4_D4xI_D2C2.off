OFF
#
# tet_4_D4xI_D2C2
#
# file generated with python script by Marcel Tunnissen
# Vertices Faces Edges
16 16 24
# Vertices
-1 0 1.41421356237309
1 1.41421356237309 0
-1 0 -1.41421356237309
1 -1.4142135623731 0
0 1 1.41421356237309
1.41421356237309 -1 0
0 1 -1.41421356237309
-1.41421356237309 -1 0
0 -1 1.41421356237309
-1.41421356237309 1 0
0 -1 -1.41421356237309
1.41421356237309 1 0
1 0 1.41421356237309
-1 -1.41421356237309 0
1 0 -1.41421356237309
-1 1.4142135623731 0
# Sides and colours
3  0 1 2  220 159 220
3  0 3 1  220 159 220
3  2 3 0  220 159 220
3  1 3 2  220 159 220
3  4 5 6  254 214 0
3  4 7 5  254 214 0
3  6 7 4  254 214 0
3  5 7 6  254 214 0
3  8 9 10  254 214 0
3  8 11 9  254 214 0
3  10 11 8  254 214 0
3  9 11 10  254 214 0
3  12 13 14  220 159 220
3  12 15 13  220 159 220
3  14 15 12  220 159 220
3  13 15 14  220 159 220
# END
# Color alternative based on D2xI
# Used colour alternative 0 (max 0)