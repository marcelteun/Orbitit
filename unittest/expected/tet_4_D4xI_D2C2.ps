%!PS-Adobe-2.0
%%Title: tet_4_D4xI_D2C2
%%Creator: PS.py written by marcelteun
%%Pages: 4 1
%%BoundingBox: 0 0 595 842
%%EndComments
%%BeginSetup
%%EndSetup
%%Page: 1 1
%%BeginPage
10.000000 10.000000 translate
0.000000 0.000000 translate
/vertices [
  [0.1498299 -1.6261051] % 0
  [1.3333333 0.942809] % 1
  [-1.4831632 0.683296] % 2
  [0.8232855 -0.1643015] % 3
  [0.1195236 0.8309682] % 4
  [-0.6582305 0.7593056] % 5
  [0.49647 -0.8736875] % 6
  [-0.8232855 0.1643015] % 7
  [-0.1195236 -0.8309682] % 8
  [-0.1195236 -0.8309682] % 9
  [0.7475469 0.5285955] % 10
  [0.7475469 0.5285955] % 11
  [-0.8232855 0.1643015] % 12
  [-0.6666667 -0.4714045] % 13
  [0.7415816 -0.341648] % 14
  [0.7415816 -0.341648] % 15
  [-0.074915 0.8130525] % 16
  [-0.074915 0.8130525] % 17
  [-0.6666667 -0.4714045] % 18
] def
/faces [
  [0 1 2][3 4][5 6][7 8][9 10][11 12][13 14][15 16][17 18]
] def
/bbox [-1.4831632 -1.6261051 1.3333333 0.942809] def
/scalingSize 200 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
568.299316 0 translate
-568.299316 518.782820 translate

showpage
%%EndPage
%%Page: 2 2
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [-1.6329932 0] % 0
  [0.8164966 1.4142136] % 1
  [0.8164966 -1.4142136] % 2
  [0.8164966 -0.4142136] % 3
  [-0.2391463 0.8047379] % 4
  [-0.9155542 0.4142136] % 5
  [0.8164966 -1] % 6
  [0.8164966 1] % 7
  [-0.9155542 -0.4142136] % 8
  [-0.2391463 -0.8047379] % 9
  [0.8164966 0.4142136] % 10
  [-0.4082483 -0.7071068] % 11
  [0.8164966 0] % 12
  [-0.4082483 0.7071068] % 13
  [-0.4082483 -0.7071068] % 14
  [0.8164966 0] % 15
  [-0.4082483 0.7071068] % 16
] def
/faces [
  [0 2 1][3 4][5 6][7 8][9 10][11 12][13 14][15 16]
] def
/bbox [-1.6329932 -1.4142136 0.8164966 1.4142136] def
/scalingSize 200 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
494.897949 0 translate
-494.897949 570.685425 translate

showpage
%%EndPage
%%Page: 3 3
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [0.1498299 1.6261051] % 0
  [-1.4831632 -0.683296] % 1
  [1.3333333 -0.942809] % 2
  [0.7475469 -0.5285955] % 3
  [-0.1195236 0.8309682] % 4
  [-0.1195236 0.8309682] % 5
  [-0.8232855 -0.1643015] % 6
  [-0.8232855 -0.1643015] % 7
  [0.7475469 -0.5285955] % 8
  [0.49647 0.8736875] % 9
  [-0.6582305 -0.7593056] % 10
  [0.1195236 -0.8309682] % 11
  [0.8232855 0.1643015] % 12
  [-0.074915 -0.8130525] % 13
  [0.7415816 0.341648] % 14
  [0.7415816 0.341648] % 15
  [-0.6666667 0.4714045] % 16
  [-0.6666667 0.4714045] % 17
  [-0.074915 -0.8130525] % 18
] def
/faces [
  [1 2 0][3 4][5 6][7 8][9 10][11 12][13 14][15 16][17 18]
] def
/bbox [-1.4831632 -0.942809 1.3333333 1.6261051] def
/scalingSize 200 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
568.299316 0 translate
-568.299316 518.782820 translate

showpage
%%EndPage
%%Page: 4 4
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [-0.8164966 1.4142136] % 0
  [1.6329932 0] % 1
  [-0.8164966 -1.4142136] % 2
  [0.2391463 0.8047379] % 3
  [-0.8164966 -0.4142136] % 4
  [-0.8164966 -1] % 5
  [0.9155542 0.4142136] % 6
  [-0.8164966 0.4142136] % 7
  [0.2391463 -0.8047379] % 8
  [0.9155542 -0.4142136] % 9
  [-0.8164966 1] % 10
  [-0.8164966 0] % 11
  [0.4082483 -0.7071068] % 12
  [0.4082483 0.7071068] % 13
  [-0.8164966 0] % 14
  [0.4082483 -0.7071068] % 15
  [0.4082483 0.7071068] % 16
] def
/faces [
  [0 2 1][3 4][5 6][7 8][9 10][11 12][13 14][15 16]
] def
/bbox [-0.8164966 -1.4142136 1.6329932 1.4142136] def
/scalingSize 200 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
494.897949 0 translate

showpage
%%EndPage

%%Tailer
%%EOF
