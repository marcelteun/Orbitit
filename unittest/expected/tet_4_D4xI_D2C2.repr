orbitit.geom_3d.OrbitShape(
    vs=[
        orbitit.geomtypes.Vec3([1, 0, 1.414213562373]),
        orbitit.geomtypes.Vec3([-1, -1.414213562373, 0]),
        orbitit.geomtypes.Vec3([1, 0, -1.414213562373]),
        orbitit.geomtypes.Vec3([-1, 1.414213562373, 0])
    ],
    fs=[
        [0, 1, 2],
        [0, 3, 1],
        [2, 3, 0],
        [1, 3, 2]
    ],
    es=[0, 1, 1, 2, 0, 2, 0, 3, 1, 3, 2, 3],
    colors=[
        [220, 159, 220],
        [254, 214, 0],
        [254, 214, 0],
        [220, 159, 220]
    ],
    final_sym=
        orbitit.isometryDnxI(
            [
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, 0, 0, 1]),
                    orbitit.geomtypes.Quat([0, 0, 0, -1]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0.707106781187, 0, 0, 0.707106781187]),
                    orbitit.geomtypes.Quat([0.707106781187, 0, 0, -0.707106781187]),
                )),
                orbitit.geomtypes.RotInv3((
                    orbitit.geomtypes.Quat([-0.707106781187, 0, 0, -0.707106781187]),
                    orbitit.geomtypes.Quat([0.707106781187, 0, 0, -0.707106781187]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 1, 0, 0]),
                    orbitit.geomtypes.Quat([0, 1, 0, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([-0.707106781187, 0, 0, 0.707106781187]),
                    orbitit.geomtypes.Quat([-0.707106781187, 0, 0, -0.707106781187]),
                )),
                orbitit.geomtypes.RotInv3((
                    orbitit.geomtypes.Quat([0.707106781187, 0, 0, -0.707106781187]),
                    orbitit.geomtypes.Quat([-0.707106781187, 0, 0, -0.707106781187]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, -1, 0, 0]),
                    orbitit.geomtypes.Quat([0, 1, 0, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([-1, 0, 0, 0]),
                    orbitit.geomtypes.Quat([-1, 0, 0, 0]),
                )),
                orbitit.geomtypes.RotInv3((
                    orbitit.geomtypes.Quat([1, 0, 0, 0]),
                    orbitit.geomtypes.Quat([-1, 0, 0, 0]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 0.707106781187, -0.707106781187, 0]),
                    orbitit.geomtypes.Quat([0, 0.707106781187, -0.707106781187, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, -0.707106781187, 0.707106781187, 0]),
                    orbitit.geomtypes.Quat([0, 0.707106781187, -0.707106781187, 0]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, -0.707106781187, -0.707106781187, 0]),
                    orbitit.geomtypes.Quat([0, -0.707106781187, -0.707106781187, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, 0.707106781187, 0.707106781187, 0]),
                    orbitit.geomtypes.Quat([0, -0.707106781187, -0.707106781187, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, 0, 1, 0]),
                    orbitit.geomtypes.Quat([0, 0, -1, 0]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 0, -1, 0]),
                    orbitit.geomtypes.Quat([0, 0, -1, 0]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 0, 0, -1]),
                    orbitit.geomtypes.Quat([0, 0, 0, -1]),
                ))
            ]
        ),
    stab_sym=
        orbitit.isometryDnCn(
            [
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([0, 1, 0, 0]),
                    orbitit.geomtypes.Quat([0, -1, 0, 0]),
                )),
                orbitit.geomtypes.Rot3((
                    orbitit.geomtypes.Quat([-1, 0, 0, 0]),
                    orbitit.geomtypes.Quat([-1, 0, 0, 0]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 0, 0, -1]),
                    orbitit.geomtypes.Quat([0, 0, 0, -1]),
                )),
                orbitit.geomtypes.Refl3((
                    orbitit.geomtypes.Quat([0, 0, 1, 0]),
                    orbitit.geomtypes.Quat([0, 0, 1, 0]),
                ))
            ]
        ),
    name='tet_4_D4xI_D2C2',
)
