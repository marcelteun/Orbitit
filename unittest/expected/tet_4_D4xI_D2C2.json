{"class": "orbit_shape", "data": {"base": {"fs": [[0, 1, 2], [0, 3, 1], [2, 3, 0], [1, 3, 2]], "vs": [[1, 0, 1.414213562373], [-1, -1.414213562373, 0], [1, 0, -1.414213562373], [-1, 1.414213562373, 0]]}, "col_alt": 0, "col_sym": "", "cols": [[220, 159, 220], [254, 214, 0]], "final_sym": {"class": "D4xI", "data": {"generator": {"axis_2": [1, 0, 0], "axis_n": [0, 0, 1], "n": 4}}}, "name": "tet_4_D4xI_D2C2", "no_of_cols": 2, "stab_sym": {"class": "D2C2", "data": {"generator": {"axis_n": [1, 0, 0], "normal_r": [0, 1, 0]}}}, "version": 2}}